### PR TITLE
[AND-922] Integrate appcore with hilt

### PR DESCRIPTION
**Pull request by kokeroulis**
- Integrate appcore with hilt
- Get requests doesn't have a body, update appcore accordingly
- Introduce `GITHUB_TOKEN` gradle property for token configuration
- Introduce `GITHUB_API_URL` gradle property for the github url
- `httpClient.request` deserialisation can now ignore remaining elements

### CODE REVIEW

File: Android/appcore-android/src/main/java/me/spjere/appcore/android/network/AgentHTTPClientImpl.kt [hunk 1]
```diff
@@ -17,11 +17,11 @@ import java.security.cert.CertificateException
 import javax.net.ssl.SSLHandshakeException
 import kotlin.coroutines.resumeWithException
 
-class HTTPClientImpl(
+class AgentHTTPClientImpl(
     private val okHttpClient: OkHttpClient,
     private val networkObserver: NetworkObserver,
     host: String
-) : HTTPClient {
+) : AgentHTTPClient {
```
Comment (ablenesi) [inline Android/appcore-android/src/main/java/me/spjere/appcore/android/network/AgentHTTPClientImpl.kt]: There is no need for the AgentHTTPClient at all, just HTTPClient should be enough.

File: Android/app/build.gradle.kts [hunk 2]
```diff
@@ -19,6 +20,9 @@ android {
         versionName = "1.0"
 
         testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+
+        buildConfigField("String", "GITHUB_API_URL", propOrDefault("GITHUB_API_URL", ""))
```
Comment (ablenesi) [inline Android/app/build.gradle.kts]: Not sure why the API needs to be from properties?

Sample should be simple

File: Android/app/src/main/java/me/sphere/unicorn/di/AppCoreModule.kt [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+@Module
+@InstallIn(SingletonComponent::class)
+object AppCoreModule {
+
+    @Provides
+    @Singleton
+    fun environment() = Environment(
+        backendUrl = BuildConfig.GITHUB_API_URL,
+        githubToken =BuildConfig.GITHUB_TOKEN
```
Comment (ablenesi) [inline Android/app/src/main/java/me/sphere/unicorn/di/AppCoreModule.kt]:
```suggestion
        githubToken = BuildConfig.GITHUB_TOKEN
```

File: Android/app/src/main/java/me/sphere/unicorn/di/AppCoreModule.kt [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    )
+
+    @Provides
+    @Singleton
+    fun agentHttpClient(
+        okHttpClient: OkHttpClient,
+        environment: Environment,
+        networkObserver: NetworkObserver
+    ): AgentHTTPClient =
+        AgentHTTPClientImpl(okHttpClient, networkObserver, environment.backendUrl)
```
Comment (ablenesi) [inline Android/app/src/main/java/me/sphere/unicorn/di/AppCoreModule.kt]: We don't have an agent, using `HttpClient` should be sufficient.

Sample should be simple :D

File: Android/app/src/main/java/me/sphere/unicorn/di/AppCoreModule.kt [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+    @Provides
+    @Singleton
+    fun logger(): Logger = Logger(true, AppCoreLoggingBackend())
```
Comment (ablenesi) [inline Android/app/src/main/java/me/sphere/unicorn/di/AppCoreModule.kt]:
```suggestion
    fun logger(): Logger = Logger(isErrorEnabled = true, AppCoreLoggingBackend())
```

**Review by ablenesi — APPROVED** [review]
Overall looks good, thanks for resolving parsing errors.

- We should get rid of AgentHttpClient, this can be done on a separate PR.

File: Android/app/src/main/java/me/sphere/unicorn/di/AppCoreModule.kt (hunk 4 above)
Comment (kokeroulis) [reply Android/app/src/main/java/me/sphere/unicorn/di/AppCoreModule.kt]: yeap ok. `Backend0StoreActorsBuilder` requires a `HttpClient` not an agent, i thought it required an agent... I will update it

**Pull request merged**